ravenframework/Samplers/CustomSampler.py

```diff
@@ -153,10 +154,16 @@ def _localGenerateAssembler(self,initDict):
     for key, val in self.dependentSample.items():
       if val not in initDict['Functions'].keys():
         self.raiseAnError(f'Function {val} was not found among the available functions: {initDict["Functions"].keys()}')
-      self.funcDict[key] = initDict['Functions'][val]
+      fPointer = namedtuple("func", ['methodName', 'instance'])
+      mName = 'evaluate'
       # check if the correct method is present
-      if "evaluate" not in self.funcDict[key].availableMethods():
-        self.raiseAnError(IOError, f'Function {self.funcDict[key].name} does not contain a method named "evaluate". It must be present if this needs to be used in a Sampler!')
+      if val not in initDict['Functions'][val].availableMethods():
+        if "evaluate" not in initDict['Functions'][val].availableMethods():
+          self.raiseAnError(IOError, f'Function {initDict["Functions"][val].name} does contain neither a method named "{val}" nor "evaluate". '
+                            'It must be present if this needs to be used in a Sampler!')
+      else:
+        mName = val
+      self.funcDict[key] = fPointer(mName, initDict['Functions'][val])
```

Comment:
Can these lines be handled in the Sampler base class?

Reply:
Unfortunately it seems that the CustomSampler and the EnsembleForward are "special" cases and they take care of they functions directly. I did not add this "exception" with this implementation but it was like that already unfortunately.